projects/packages/stats/src/class-xmlrpc.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * The Stats XMLRPC class
+ *
+ * @package automattic/jetpack-stats
+ */
+
+namespace Automattic\Jetpack\Stats;
+
+use Automattic\Jetpack\Constants;
+
+/**
+ * Stats XMLRPC.
+ *
+ * Adds additional methods to the WordPress XML-RPC API for handling Stats specific features.
+ *
+ * @since $$next-version$$
+ */
+class XMLRPC {
+
+	/**
+	 * Singleton XMLRPC instance.
+	 *
+	 * @var XMLRPC
+	 **/
+	private static $instance = null;
+
+	/**
+	 * Private XMLRPC constructor.
+	 *
+	 * Use the XMLRPC::init() method to get an instance.
+	 */
+	private function __construct() {
+		add_filter( 'jetpack_xmlrpc_unauthenticated_methods', array( $this, 'xmlrpc_methods' ), 10, 3 );
+	}
+
+	/**
+	 * Initialize class and get back a singleton instance.
+	 *
+	 * @return XMLRPC
+	 */
+	public static function init() {
+		if ( null === self::$instance ) {
+			self::$instance = new XMLRPC();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Adds additional methods to the WordPress xmlrpc API for handling Stats specific features
+	 *
+	 * @param array $methods The Jetpack API methods.
+	 *
+	 * @return array
+	 */
+	public function xmlrpc_methods( $methods ) {
+
+		$methods['jetpack.getBlog'] = array( $this, 'get_blog' );
+
+		return $methods;
+	}
+
+	/**
+	 * Stats Get Blog.
+	 *
+	 * @return array
+	 */
+	public function get_blog() {
```

Comment:
It is kinda random, and not sure if it makes sense. 

Assuming every `get_option` call will query the DB, and there is no caching added, I wonder if this method could lead to some performance issues. Also, it seems that the returned value wouldn't change between calls.

Reply:
Good point. TBH at this point I'm trying to slowly refactor the [stats module code](https://github.com/Automattic/jetpack/blob/d38405b569aad92c4164b61acb57b884dc11b2c0/projects/plugins/jetpack/modules/stats.php) by bringing basic pieces of functionality to the Stats package.
There are lots of things that I'd like to change, however trying to draw a reasonable line and find a balance between refactoring, ensuring backwards compatibility and making quick progress.
All this to prepare you for future reviews :)
For now, I hope it's ok to add a todo here to add caching in a follow up PR.
In general, I've only found this endpoint being used by the Jetpack Debugger so I wouldn't be too concerned about performance issues under this context.
Many thanks for taking a look!